types/types.cc

```diff
@@ -844,6 +846,36 @@ bool abstract_type::references_user_type(const sstring& keyspace, const bytes& n
     return find(*this, visitor{keyspace, name});
 }
 
+std::unordered_set<sstring> user_type_impl::get_all_referenced_user_types() const {
+    struct visitor {
+        std::unordered_set<sstring> referenced_udts;
+
+        void operator()(const abstract_type&) { return; }
+        void operator()(const reversed_type_impl& r) {
+            visit(*r.underlying_type(), *this);
+        }
+        void operator()(const user_type_impl& u) {
+            referenced_udts.insert(u.get_name_as_string());
```

Comment:
Shouldn't we recurse here? Or is it purposefully a shallow visitor?

Reply:
It is purposefully shallow. To be fair, the name of the function and the comment near it should mention it, but they don't.